src/main/webapp/app/overview/page-discussion-section/page-discussion-section.component.ts

```diff
@@ -0,0 +1,161 @@
+    course?: Course;
+    collapsed = false;
+    createdPost: Post;
+    posts: Post[];
+    isLoading = true;
+    readonly pageType = PageType.PAGE_SECTION;
+
+    private postsSubscription: Subscription;
+    private paramSubscription: Subscription;
+
+    constructor(private metisService: MetisService, private activatedRoute: ActivatedRoute, private courseManagementService: CourseManagementService) {}
+
+    /**
+     * on initialization: initializes the metis service, fetches the posts for the exercise or lecture the discussion section is placed at,
+     * creates the subscription to posts to stay updated on any changes of posts in this course
+     */
+    ngOnInit(): void {
+        this.paramSubscription = this.activatedRoute.params.subscribe((params) => {
+            const courseId = parseInt(params['courseId'], 10);
+            this.courseManagementService.findOneForDashboard(courseId).subscribe((res: HttpResponse<Course>) => {
+                if (res.body !== undefined) {
+                    this.course = res.body!;
+                    this.metisService.setCourse(this.course!);
+                    this.metisService.setPageType(this.pageType);
+                    this.metisService.getFilteredPosts({
+                        exerciseId: this.exercise?.id,
+                        lectureId: this.lecture?.id,
+                    });
+                    this.createEmptyPost();
+                }
+            });
+        });
+        this.postsSubscription = this.metisService.posts.pipe(map((posts: Post[]) => posts.sort(this.sectionSortFn))).subscribe((posts: Post[]) => {
+            this.posts = posts;
+            this.isLoading = false;
+        });
+    }
+
+    /**
+     * on changes: fetches the posts for the exercise or lecture the discussion section is placed at,
+     * triggers the creation of an empty post
+     */
+    ngOnChanges(): void {
+        this.createEmptyPost();
+    }
+
+    /**
+     * on leaving the page, the modal should be closed, subscriptions unsubscribed
+     */
+    ngOnDestroy(): void {
+        this.paramSubscription?.unsubscribe();
+        this.postsSubscription?.unsubscribe();
+        this.postCreateEditModal?.modalRef?.close();
+    }
+
+    /**
+     * sorts posts by following criteria
+     * 1. criterion: displayPriority is PINNED -> pinned posts come first
+     * 2. criterion: displayPriority is ARCHIVED  -> archived posts come last
+     * -- in between pinned and archived posts --
+     * 3. criterion: vote-emoji count -> posts with more vote-emoji counts comes first
+     * 4. criterion: creationDate -> most recent comes at the end (chronologically from top to bottom)
+     * @return Post[] sorted array of posts
+     */
+    sectionSortFn(postA: Post, postB: Post): number {
+        if (postA.displayPriority === DisplayPriority.PINNED && postB.displayPriority !== DisplayPriority.PINNED) {
```

Comment:
true, good catch, I changed the description

Reply:
I see. I tried to think of a solution that unifies the two functions, but failed to find a satisfactory way of mixing sorting votes descendingly and dates ascendingly. For the future it might make sense to separate the compare functions. Each function could have an optional "and then" sorter that gets called if the current comparison returns 0. That way both functions could be expressed as a simple chain of such basic sorters